### PR TITLE
Creates and Organizes Reference Section

### PR DESCRIPTION
**Pull request by victorhuangwq**
## Description

This creates a new section in the documentation, that contains all the APIs and CLIs that Autolab has. 

## Motivation and Context

All the references that Autolab has for APIs and CLIs are currently scattered around the whole documentation.
This helps consolidate them into one area, making it easier for developers to find and use.

## How Has This Been Tested?

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation, which is located at [Autolab Docs](https://github.com/autolab/docs)
- [ ] I have updated the documentation accordingly, included in this PR

## Other issues / help required

If unsure, feel free to submit first and we'll help you along.

### CODE REVIEW

**Review by ashleyzhang — CHANGES_REQUESTED** [review]
Just a couple small wording changes but good otherwise! I like the new structure.

File: mkdocs.yml [hunk 1]
```diff
@@ -22,17 +22,18 @@ nav:
...
-    - Autolab API:
-        - Overview: api-overview.md
-        - Reference: api-interface.md
-        - Command Line Interface: command-line-interface.md
-        - Managing Authorized Apps: api-managing-authorized-apps.md
+    - Reference: 
+        - Reference Documentation: reference.md
+        - Autolab Frontend API:
+            - Overview: api-overview.md
+            - Reference: api-interface.md
```
Comment (xinyis991105) [inline mkdocs.yml:29]: Thought: For Reference/Autolab Frontend API/Reference, would it be better if we could name it Endpoints or something other than Reference? The nested "Reference" feels a bit redundant to me. But I'm up to the majority's opinion.

Comment (victorhuangwq) [reply mkdocs.yml:29]: I think endpoints is a suitable change

File: docs/reference.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Reference documentation
+
+This section includes all the references for the Autolab platform's various Application Programming Interfaces (APIs) and Command Line Interfaces (CLIs).
```
[the remark this answers is not in the document]
Comment (victorhuangwq) [reply docs/reference.md:3]: suggestion taken

File: docs/tango-vmms.md [hunk 3]
```diff
@@ -1,4 +1,9 @@
-This page documents the interface for Tango's Virtual Machine Management Systems' (VMMSs) API and instructions for setting up VMMSs. See [the vmms directory](https://github.com/autolab/Tango/tree/master/vmms) in Tango for example implementations.
+This page documents the interface for Tango's Virtual Machine Management Systems' (VMMSs) API and instructions for setting up VMMSs. 
+
+See [the vmms directory](https://github.com/autolab/Tango/tree/master/vmms) in Tango for example implementations. 
+
+We currently provide [Docker VMMS](/tango/#docker-vmms-setup) and the [Amazon EC2 VMMS](tango/#amazon-ec2-vmms-setup) interfaces. The following API is for developers would intend to use a different kind of VMMS for Tango's autograding purposes.
```
[the remark this answers is not in the document]
Comment (victorhuangwq) [reply docs/tango-vmms.md:5]: suggestion taken

File: docs/reference.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashleyzhang) [reply docs/reference.md:3]: Were you able to change the wording here? Not a big deal but looks like it wasn’t changed. Or let me know if you decided it’s better as is and I’ll approve it!

**Review by ashleyzhang — CHANGES_REQUESTED** [review]

**Comment by victorhuangwq** [comment]
@ashleyzhang oops sorry I didn't save the changes for that update. I have since updated it.

**Review by ashleyzhang — APPROVED** [review]
LGTM!

**Review by xinyis991105 — APPROVED** [review]
LGTM!

**Pull request merged**